xcell/cls/cl.py

```diff
@@ -112,22 +121,24 @@ def get_nmt_fields(self):
         f2 = mapper2.get_nmt_field()
         return f1, f2
 
-    def get_workspace(self):
+    def get_workspace(self, read_unbinned_MCM=True):
         if self._w is None:
-            self._w = self._compute_workspace()
+            self._w = \
+                self._compute_workspace(read_unbinned_MCM=read_unbinned_MCM)
         return self._w
 
     def get_workspace_cov(self):
         if self._wcov is None:
             spin0 = self.data.data['cov'].get('spin0', False)
             if spin0 and (self.get_spins() != (0, 0)):
-                self._wcov = self._compute_workspace(spin0=spin0)
+                self._wcov = self._compute_workspace(spin0=spin0,
+                                                     read_unbinned_MCM=False)
```

Comment:
why do we call `_compute_workspace` here instead of `get_workspace` as we do below?

Reply:
Because `get_workspace` is the method to get `self._w`.

Reply:
Wait, we're calling `get_workspace` just below to get `self._wcov`

Reply:
Oh, wait, `get_workspace` doesn't have a `spin0` option. I think I get it.

Reply:
Exactly. You use get_workspace in that case because you are not using the spin0 approx.